chapter2/demands.md

```diff
@@ -0,0 +1,97 @@
+# Demands
+
+When link between two Membrane elements works in `:pull` mode, data between them is sent only when requested by the upstream element. That "request" in our framework is called "demand".
```

Comment:
I think this chapter is missing info about what happens under the hood - PullBuffers, their minimal demand and so on

Reply:
It can be added in a separate PR ;)  issue #13